writeengine/splitter/we_brmupdater.cpp

```diff
@@ -353,6 +353,15 @@ bool WEBrmUpdater::prepareCasualPartitionInfo()
 
             pTok = strtok(NULL, " ");
 
+            if (pTok)
+                cpInfoMerge.colWidth = boost::lexical_cast<int32_t>(pTok);
+            else
+            {
+                //cout << "CP Entry : " << aEntry << endl;
+                throw (runtime_error("Bad column width in CP entry string"));
+            }
+            pTok = strtok(NULL, " ");
+
```

Comment:
We need to add support for 128-bit min/max values as well here. The sending side `BRMReporter::sendCPToFile` either streams `CPInfoMerge::max/CPInfoMerge::min` or `CPInfoMerge::bigMax/CPInfoMerge::bigMin` based on `CPInfoMerge::type` and `CPInfoMerge::colWidth`. So the parser here should first parse out `type` and `colWidth` and then based on these 2 values, should either set the 128-bit or the 64-bit min/max values.

Reply:
Thanks!

It is not possible to have tests for that functionality using joins. I've created task specifically to address your comment: https://jira.mariadb.org/browse/MCOL-4652